app/src/main/java/com/lastaoutdoor/lasta/data/db/UserDBRepositoryImpl.kt

```diff
@@ -124,6 +124,58 @@ class UserDBRepositoryImpl @Inject constructor(context: Context, database: Fireb
     userDocumentRef.update("favorites", FieldValue.arrayUnion(activityId)).await()
   }
 
+  override suspend fun getUsersByUsernameWithSubstring(
+      query: String,
+      friends: List<UserModel>,
+      user: UserModel
+  ): List<UserModel> {
+    // if the query is empty, return an empty list
+    if (query.isEmpty() || query.isBlank()) {
+      return emptyList()
+    }
+    // get all users that are not the current user or his/her friends
+    val allUsersUsernames =
+        userCollection.whereNotIn("email", friends.map { it.email }.plus(user.email)).get().await()
+    if (allUsersUsernames.isEmpty) {
+      return emptyList()
+    }
+    val usersToList = mutableListOf<UserModel>()
+    // for each user, check if the username contains the query
+    allUsersUsernames.documents.forEach { doc ->
+      // get the username of the user
+      val userName = doc.getString("userName") ?: ""
+      // if the username contains the query, return the user
+      if (userName.lowercase().contains(query)) {
```

Comment:
It's a nice idea to force the usernames to lower case, but therefore it should also be the same for the query variable. Maybe it is done when calling this function, but if we reuse it, it would be more practical and more mistake proof to add a query = query.lowercase()

Reply:
You're right, for other calls it's a good thing to do